calico_containers/calicoctl.py

```diff
@@ -356,6 +356,39 @@ def node(ip, node_image, log_dir, ip6="", as_num=None):
     # Print warnings for any known system issues before continuing
     checksystem(fix=False, quit_if_error=False)
 
+    # Get IP address of host, if none was specified
+    if not ip:
+        ip = get_first_ip()
+        if ip:
+            print "No IP provided. Using detected IP: %s" % ip
+        else:
+            print "Couldn't autodetect a management IP address. Please rerun" \
+                  "command with the --ip=<IP_ADDRESS> flag."
+            sys.exit(1)
+
+    # Warn if this hostname is already in use.
+    # If there's already a calico-node container on this host, they're probably
+    # just re-running node to do an update, so only issue a warning
+    if len(docker_client.containers(filters={'name': 'calico-node'})) == 0:
+        # Otherwise, check if another host with the same hostname
+        # is already configured
+        try:
+            current_ipv4, current_ipv6 = client.get_host_ips(hostname)
+        except KeyError:
+            # No other machine has registered configuration under this hostname.
+            # This must be a new host with a unique hostname, which is the
+            # expected behavior.
+            pass
+        else:
+            if current_ipv4 != "" and current_ipv4 != ip:
+                # User is either reconfiguring this node to use another IP,
+                # or another host may share the same hostname.
+                print "WARNING: Hostname %s is already in use with IP address " \
+                      "%s. Calico requires each compute host to have a " \
+                      "unique hostname. If this is your first time running" \
+                      "'calicoctl node' on this host, check to make sure" \
+                      "another host is not already using the same hostname"  % (hostname, ip)
+
```

Comment:
move these procedures into their own functions.  `node()` is too cluttered as it is!